WordPress/gutenberg
# Site Editor: Remove unused 'NavigateToLink' component.

## What?
Closes #27165.

PR removes the unused `NavigateToLink` component from the `edit-site` package.

## Why?
The component hasn't been used since #47454.

## Testing Instructions
1. Open the Site Editor.
2. Confirm it loads without errors.
